### PR TITLE
Fix unused arg in PSBT impl_psbt_get_pair macro

### LINKED ISSUE #754: Remove unused last argument in `impl_psbt_get_pair` macro
See details in https://github.com/rust-bitcoin/rust-bitcoin/pull/669#discussion_r777062798

### PR DESCRIPTION
**Pull request by dr-orlovsky**
Closes #754

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK e8a06fd41d29607f7eca5074581d010fe5428d37

File: src/util/psbt/macros.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ macro_rules! impl_psbt_insert_pair {
 
 #[cfg_attr(rustfmt, rustfmt_skip)]
 macro_rules! impl_psbt_get_pair {
-    ($rv:ident.push($slf:ident.$unkeyed_name:ident as <$unkeyed_typeval:expr, _>|<$unkeyed_value_type:ty>)) => {
+    ($rv:ident.push($slf:ident.$unkeyed_name:ident as Option<$unkeyed_typeval:ident>)) => {
```
Comment (sanket1729) [inline src/util/psbt/macros.rs:128]: nit: Remove the `Option` everywhere?

File: src/util/psbt/macros.rs [hunk 2]
```diff
@@ -136,7 +136,7 @@ macro_rules! impl_psbt_get_pair {
             });
         }
     };
-    ($rv:ident.push($slf:ident.$keyed_name:ident as <$keyed_typeval:expr, $keyed_key_type:ty>|<$keyed_value_type:ty>)) => {
+    ($rv:ident.push($slf:ident.$keyed_name:ident as BTreeMap<$keyed_typeval:ident>)) => {
```
Comment (sanket1729) [inline src/util/psbt/macros.rs:139]: We can also remove the BTreeMap here, and change this to `rv.push` and the above can be `rv.push_map`. The above types `Option` and `BTreeMap` are merely matching the syntax and not used in the macro.

If you do the above suggestions, the code might look simpler:

```rust
            rv.push(self.witness_script as PSBT_OUT_WITNESS_SCRIPT)
            rv.push_map(self.bip32_derivation as PSBT_OUT_BIP32_DERIVATION)
```

**Review by sanket1729 — COMMENTED** [review]
Suggested another style which I think is probably a bit clearer and cleaner.

**Comment by dr-orlovsky** [comment]
@sanket1729 simplified as you suggested - also removed unnecessary/confusing `as` part

**Review by apoelstra — APPROVED** [review]
ACK 1b77e3609c84e410bbea091b92341f9863fb59a5

Good idea @sanket1729 !

**Review by RCasatta — APPROVED** [review]
utACK 1b77e3609c84e410bbea091b92341f9863fb59a5

much simpler

**Comment by apoelstra** [comment]
Let's get one more ACK just in case others have opinions about these macros.

**Review by Kixunil — APPROVED** [review]
ACK 1b77e3609c84e410bbea091b92341f9863fb59a5

Potentially for future nit PR rename `$slf` to `$this` which should be more readable.

**Pull request merged**